packages/workflow/src/utils.ts

```diff
@@ -28,7 +28,7 @@ export const deepCopy = <T extends ((object | Date) & { toJSON?: () => string })
 		return clone as T;
 	}
 	// Object
-	const clone = {} as T;
+	const clone = Object.create(Object.getPrototypeOf(source));
```

Comment:
we might have the same issue here: the change makes semantic sense, but since we did not include the prototype in the cloned objects before, it's hard to know what side-effects this addition would have. Can you please remove this? 
we should definitely do this, just not in this PR.

Reply:
It works with `Object.create(null)` as well and passing the tests but in my perf testing it fails and works only with `Object.create(Object.getPrototypeOf(source))`

Reply:
maybe it's just `assert.deepStrictEqual` in nodejs

Reply:
this works in both places `Object.create(Object.getPrototypeOf({}));`